### PR TITLE
Added Modal for word detials in word-list-component

### PR DESCRIPTION
**Pull request by mahaelbasuony**
Ref #42

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { AddWordComponent, WordListComponent } from './components';
 import { WordService, ValidationService } from './services';
 import { MultiComponent, ErrorMessageComponent } from './shared/ui-components';
 
+import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
```
Comment (medhatdawoud) [inline src/app/app.module.ts]: Is this an official library from angular ?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -24,7 +25,8 @@ import { MultiComponent, ErrorMessageComponent } from './shared/ui-components';
     HttpModule,
     FormsModule,
     NgReduxModule,
-    ReactiveFormsModule
+    ReactiveFormsModule,
+    NgbModule.forRoot()
```
Comment (medhatdawoud) [inline src/app/app.module.ts]: What is the purpose of this line ?

File: src/app/components/word-list/word-list.component.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div class="words-list tile-view">
-  <div *ngFor="let w of words$ | async" class="one-word">
+  <div *ngFor="let w of words$ | async" class="one-word" (click)="wordDetialsModal(w,content)">
```
Comment (medhatdawoud) [inline src/app/components/word-list/word-list.component.html]: Do you have to add content each time you click? or you can remove it from the function and use it in the code ?

File: src/app/app.module.ts (hunk 1 above)
Comment (mahaelbasuony) [reply src/app/app.module.ts]: https://ng-bootstrap.github.io/#/getting-started

File: src/app/app.module.ts (hunk 2 above)
Comment (mahaelbasuony) [reply src/app/app.module.ts]: it's important for modal appearance 
https://ng-bootstrap.github.io/#/getting-started

File: src/app/components/word-list/word-list.component.html (hunk 3 above)
Comment (mahaelbasuony) [reply src/app/components/word-list/word-list.component.html]: fixed 
https://ng-bootstrap.github.io/#/components/modal/examples

**Pull request merged**